src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,186 @@
+package duke.task;
+
+import duke.Parser;
+import duke.Ui;
+
+import java.time.LocalDateTime;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Encapsulates an arraylist of Task with the various methods
+ */
+public class TaskList {
+    private ArrayList<Task> list;
+
+    public TaskList(ArrayList<Task> list) {
+        this.list = list;
+    }
+
+    public ArrayList<Task> getList() {
+        return this.list;
+    }
+
+    /**
+     * Nicely prints out all the task
+     */
+    public String list() {
+        String res = "Here are the tasks in your list:\n";
+        for (int i = 0; i < list.size(); i++) {
+            res += i + 1 + ". " + list.get(i) +  "\n";
+        }
+        return res;
+    }
+
+    /**
+     * Prints out default message after adding a dask
+     */
+    public String addTask() {
+        return "Got it. I've added this duke.task:\n" + list.get(list.size() - 1)
+                + "\n" + "Now you have " + list.size() + " tasks in the list.";
+    }
+
+    /**
+     * Remove a task
+     * @param i index of task in arraylist
```

Comment:
it would be nice if you could add @return to the javadoc and explain what is being returned